### DISCUSSION TITLE
Is it more efficient to include "where" condition in query with RLS?

### DISCUSSION THREAD

**Discussion by tawjaw** [supabase/supabase#791] [Questions] [answered]
Hello, 

If I have a policy on select for a table `(auth.uid() = user_id)`

this
```js
supabase.from('table_name').select('*')
```

and this 
```js
supabase.from('table_name').select('*').eq('user_id', user_id)
```

should return the same rows. But is including the equal condition makes the query more efficient in any way?

**Comment by steve-chavez** [comment, marked as answer]
Hey @tawja,

You can always check the postgres planning/execution time with `EXPLAIN ANALYZE` on the SQL Editor.

```sql
explain analyze select * from table_name where user_id = <val>;
-- Planning Time: 0.312 ms
-- Execution Time: 0.116 ms
```

For your question, RLS acts like an implicit where, so you'd be essentially doing:

```sql
select * from table_name where user_id = auth.uid() and user_id = <val>;
```

Which should give you more or less the same performance.

**Reply by tawjaw** [reply]
How can I execute a query from the sql editor as a specific user?

To run the analyze with specific auth.uid()

**Reply by steve-chavez** [reply]
Hm, I think that cannot be done from the Supabase SQL Editor for now. But you can do it through another pg client, like `psql`.

The main idea is to switch the role inside a transaction:

```sql
begin;

-- switch role
set local role authenticated;

-- set the jwt attributes 

-- this is the auth.uid() value
set local "request.jwt.claim.sub" to <val>;

-- do the query you want to test
explain analyze select * from table_name;
-- you should see the filtered rows(maybe index scan) from the RLS policy

commit;
```

**Reply by tawjaw** [reply]
Thanks for the help @steve-chavez 
I'll try that out
